source/rules/rxjsSuffixSubjectsRule.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/rxjs-tslint-rules
+ */
+/* tslint:disable:no-use-before-declare */
+
+import { tsquery } from "@phenomnomnominal/tsquery";
+import * as Lint from "tslint";
+import * as ts from "typescript";
+import { couldBeType } from "../support/util";
+
+const defaultTypesRegExp = /^EventEmitter$/;
+
+export class Rule extends Lint.Rules.TypedRule {
+  public static metadata: Lint.IRuleMetadata = {
+    description:
+      "Ensures subjects are suffixed with 'Subject'.",
+    options: {
+      properties: {
+        functions: { type: "boolean" },
+        methods: { type: "boolean" },
+        parameters: { type: "boolean" },
+        properties: { type: "boolean" },
+        suffix: { type: "string" },
+        types: { type: "object" },
+        variables: { type: "boolean" }
+      },
+      type: "object"
+    },
+    optionsDescription: Lint.Utils.dedent`
+      An optional object with optional \`functions\`, \`methods\`, \`parameters\`,
+      \`properties\` and \`variables\` properties.
+      The properies are booleans and determine whether or not subjects of that particular expression need a suffix.
+      \`parameters\`, \`properties\` and \`variables\` default to \`true\`.
+      The object also has optional \`types\` properties which are themselves
+      objects containing keys that are regular expressions and values that are booleans -
+      indicating whether suffixing is required for particular types.`,
+    requiresTypeInfo: true,
+    ruleName: '"rxjs-suffix-subjects"',
+    type: "style",
+    typescriptOnly: true
+  };
+
+  private SUFFIX: string;
+  private types: { regExp: RegExp; validate: boolean }[] = [];
+  private validate = {
+    functions: false,
+    methods: false,
+    parameters: true,
+    properties: true,
+    variables: true
+  };
+  public readonly FAILURE_MESSAGE = (identifier: string) =>
+    `Subject '${identifier}' must be suffixed with '${this.SUFFIX}'.`
+
+  public applyWithProgram(
+    sourceFile: ts.SourceFile,
+    program: ts.Program
+  ): Lint.RuleFailure[] {
+    const failures: Lint.RuleFailure[] = [];
+    const typeChecker = program.getTypeChecker();
+
+    // get configurations
+    const { ruleArguments } = this.getOptions();
+    const [options] = ruleArguments;
+    if (options) {
```

Comment:
This looks to me like the `options` must always be specified in the `tslint.json` configuration. Ideally, the options should be optional - that's what's in the rule's metadata at the top of the file.

Have a look at the implementation of some of the other rules. They should how default options can be applied.

Also, I would write two separate test fixtures: one with the options specified (as in your test); and one in which there are no options specified and the defaults are relied upon.